### PR TITLE
Add shop tooltips for total credits and ship names

### PR DESCRIPTION
**Pull request by Amazinite**
**Feature**

This PR addresses the feature described in issues #3203 and #5765.

## Summary
This PR adds new tooltips to the two shop panels (the outfitter and shipyard).
* When hovering over your number of credits, a tooltip will provide the full unabbreviated amount of credits that you have.
* When hovering over one of your ships, a tooltip will provide the name of that ship.

## Screenshots
Credits tooltip.
image
Ship name tooltip.
image
Ship name + flight check warning.
image

## Testing Done
See screenshots.

### CODE REVIEW

**Comment by TomGoodIdea** [comment]
When looking at the first two screenshots, I think dynamic tooltip background width for texts shorter than the current width would be nice.

File: source/ShopPanel.cpp [hunk 1]
```diff
@@ -907,6 +920,33 @@ void ShopPanel::DrawButtons()
...
 	}
+
+	// Draw the tooltip for your full number of credits.
+	const Rectangle creditsBox = Rectangle::FromCorner(creditsPoint, Point(SIDEBAR_WIDTH - 20, 15));
+	if(creditsBox.Contains(hoverPoint))
+		hoverCount += hoverCount < HOVER_TIME;
+	else if(hoverCount)
+		--hoverCount;
+
+	if(hoverCount == HOVER_TIME)
```
[the remark this answers is not in the document]
Comment (Amazinite) [reply source/ShopPanel.cpp:931]: > mb this means it even moves around with your mouse?

Yeah. That's how tooltips normally work. The only exception is actually the ship name/warning tooltip.

**Review by Hurleveur — APPROVED** [review]
Looks good to me code wise (tho the new version should probably be tested if it wasnt already, just to be sure)

**Comment by Amazinite** [comment]
I've updated the screenshots to reflect the latest changes.

**Comment by TomGoodIdea** [comment]
You can remove warningPoint declaration from the header since it became unused.

**Pull request merged**